woocommerce/woocommerce-ios
# [Order Creation] bottom sheet for button and totals on order form

Part of: #11467
Merge after: #11490 


## Description


As part of improving the Order Form on tablets, we are improving the use of vertical space in the Order Form.

To ensure clarity on the larger screen, we want to ensure that the total and CTA button are always visible, regardless of how many products are added to the order. This PR moves the totals and button to a new expandable sheet, as a move towards that goal.

This PR does not make the screen fully match the designs – here's a list of what's still to do in a follow-up PR:

1. Show Order Total in the collapsed state
2. Show dividers between the button, order total, and other totals, but not at the top of the view
3. Style the `Add shipping/Add Coupon` buttons as foreground rows
4. Remove the navigation bar `Done` button in edit mode
5. Landscape safe area inset support for the panel

Additionally note that although these changes are intended to benefit the tablet experience, they're primarily related to the phone until we make a split view for the Order Form in a future release. They will still work on the large modals that the tablet uses, but it's not the eventual intention.

The interuptible swipe animation is not perfect, but it's the best I can get it for the moment and reasonably reliable.

## Testing instructions


1. Launch the app
2. Navigate to the `Orders` tab
3. Tap `+`
4. Observe that the Collect Payment button is shown in a bottom sheet
6. Tap the chevron; observe that the sheet expands to show the Order Total
7. Tap it again; observe that the sheet collapses to only show the button
8. Swipe the sheet up; observe that it expands
9. Add a product; observe that the sheet expands to fit its new content

Repeat the above test for editing an order

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/e0de6c06-438e-452c-ba3f-f7b881882efc


---
